frontend/src/components/search/SearchFilters.tsx

```diff
@@ -13,36 +14,60 @@ import {
 import SearchOpportunityStatus from "src/components/search/SearchOpportunityStatus";
 import { AgencyFilterAccordion } from "./SearchFilterAccordion/AgencyFilterAccordion";
 
-export default function SearchFilters({
+const defaultFacetCounts = {
+  funding_instrument: {},
+  applicant_type: {},
+  agency: {},
+  funding_category: {},
+  opportunity_status: {},
+};
+
+export default async function SearchFilters({
   fundingInstrument,
   eligibility,
   agency,
   category,
   opportunityStatus,
+  searchResultsPromise,
 }: {
   fundingInstrument: Set<string>;
   eligibility: Set<string>;
   agency: Set<string>;
   category: Set<string>;
   opportunityStatus: Set<string>;
+  searchResultsPromise: Promise<SearchAPIResponse>;
 }) {
   const t = useTranslations("Search");
   const agenciesPromise = getAgenciesForFilterOptions();
 
+  let searchResults;
+  try {
+    searchResults = await searchResultsPromise;
+  } catch (e) {
+    console.error("Search error, cannot set filter facets", e);
+  }
+
+  const facetCounts = searchResults?.facet_counts || defaultFacetCounts;
```

Comment:
the tricky piece here is that we have an async call in this component, but the component is not suspended. I did it this way because all we care about from the search response is the facet count, which is not vitally important to the functionality of the page. If we suspended on this, it would make the entire filter section unusable until the search results load, as fallbacks (I learned while working on this) cannot be interactive.

Also note that the promise is resolved here since beyond this in the tree we are in client component territory.

This works fine as far as I can tell, but makes testing tricky, as the test suite expects things like this to be suspended

Reply:
I was thinking we'd have to do something like this, since we don't want all the filers and results flashing every time someone visits the page. This seems like a good outcome. It helps that most of the filters are hidden by default. The initial page render might be cached by a CDN at some point as well.